src/api/chainApi/rpcClient.ts

```diff
@@ -0,0 +1,76 @@
+import { ApiPromise, WsProvider } from "@polkadot/api";
+import { formatBalance } from "@polkadot/util";
+import chains, { Chain, defaultChain } from "../../chains";
+
+export class RpcClient {
+  public api?: ApiPromise;
+
+  public isApiConnected: boolean = false;
+  public isApiInitialized: boolean = false;
+  public apiError?: string;
+  public chainData: any = undefined;
+  public currentChain: Chain;
```

Comment:
currentChain stored on main provider level. To avoid confusion - replace with rpcEndpoint which is used